src/libtriton/CMakeLists.txt

```diff
@@ -22,7 +22,8 @@ endif()
 # Global UNIX CXX Flags
 if(CMAKE_COMPILER_IS_GNUCXX OR ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
     # Flags
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Wno-unused-parameter -Wno-unknown-pragmas -Wno-mismatched-tags")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -Wno-unused-parameter -Wno-unknown-pragmas")
+    # set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-mismatched-tags")
```

Comment:
What is the version of clang you are using? Because it is likely to break another clang configuration, isn't it?

Reply:
Thanks for reviewing.
I did NOT tested my changes in clang.

`-Wno-mismatched-tags` is required for clang?
It seems `-Wno-mismatched-tags` is obsoleted in my gcc (Arch Linux, Ubuntu).

Reply:
I tried compile my Triton with clang!
Here's my result.

### clang 6.0 (Arch Linux)
version info:

```
K_atc% clang --version
clang version 6.0.0 (tags/RELEASE_600/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

I built [my Triton](https://github.com/K-atc/Triton/tree/dev-fix-errors) with clang 6.0.
I encountered following compile error.

```
In file included from /home/katc/project/pin-2.14-71313-gcc.4.4.7-linux/source/tools/Triton/src/libtriton/bindings/python/init.cpp:8:
In file included from /home/katc/project/pin-2.14-71313-gcc.4.4.7-linux/source/tools/Triton/src/libtriton/includes/triton/pythonBindings.hpp:11:
In file included from /usr/include/python2.7/Python.h:88:
/usr/include/python2.7/unicodeobject.h:534:5: error: 'register' storage class specifier is
      deprecated and incompatible with C++17 [-Werror,-Wdeprecated-register]
    register PyObject *obj,     /* Object */
    ^~~~~~~~~
/usr/include/python2.7/unicodeobject.h:553:5: error: 'register' storage class specifier is
      deprecated and incompatible with C++17 [-Werror,-Wdeprecated-register]
    register PyObject *obj      /* Object */
    ^~~~~~~~~
/usr/include/python2.7/unicodeobject.h:575:5: error: 'register' storage class specifier is
      deprecated and incompatible with C++17 [-Werror,-Wdeprecated-register]
    register const wchar_t *w,  /* wchar_t buffer */
    ^~~~~~~~~
/usr/include/python2.7/unicodeobject.h:593:5: error: 'register' storage class specifier is
      deprecated and incompatible with C++17 [-Werror,-Wdeprecated-register]
    register wchar_t *w,        /* wchar_t buffer */
    ^~~~~~~~~
```

It seems `-Wno-deprecated-register` is required.

### clang 3.8 (Ubuntu 16.04)
version info:
```
tomori@nao:~/bin/pin-2.14-71313-gcc.4.4.7-linux/source/tools/Triton/build$ clang --version 
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

I built [my Triton](https://github.com/K-atc/Triton/tree/dev-fix-errors) with clang 3.8.
I encountered following compile error.

```
/home/tomori/bin/pin-2.14-71313-gcc.4.4.7-linux/source/tools/Triton/src/libtriton/includes/triton/registers_e.hpp:48:3: error: 
      'hash' defined as a class template here but previously declared as a struct template
      [-Werror,-Wmismatched-tags]
  class hash<triton::arch::registers_e>: public hash<uint64_t> {
```

It seems `-Wno-mismatched-tags` is required.

### Patch
Finally, I made following patch, and succeeded to compile in both clang 6.0 & 3.8.
(This patch is NOT applied to this pull request.)

```diff
diff --git a/src/libtriton/CMakeLists.txt b/src/libtriton/CMakeLists.txt
index 25acb8d6..037b1700 100644
--- a/src/libtriton/CMakeLists.txt
+++ b/src/libtriton/CMakeLists.txt
@@ -41,6 +41,12 @@ if(CMAKE_COMPILER_IS_GNUCXX OR ${CMAKE_CXX_COMPILER_ID} MATCHES "Clang")
     if (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 8.1)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=cast-function-type")
     endif()
+    if(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.0) # Really?
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-mismatched-tags")
+    endif()
+    if(${CMAKE_CXX_COMPILER_ID} MATCHES "Clang" AND NOT(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 6.0))
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-register")
+    endif()
 endif()
 
 # 32-bits
```

Reply:
I got why Travis CI fails! I'll update my pullreq.